components/ai_chat/core/common/pref_names.cc

```diff
@@ -45,15 +45,18 @@ void RegisterProfilePrefsForMigration(PrefRegistrySimple* registry) {
 }
 
 void MigrateProfilePrefs(PrefService* profile_prefs) {
-  profile_prefs->ClearPref(kObseleteBraveChatAutoGenerateQuestions);
-  // migrate model key from "chat-default" to "chat-basic"
-  static const std::string kDefaultModelBasicFrom = "chat-default";
-  static const std::string kDefaultModelBasicTo = "chat-basic";
-  if (auto* default_model_value =
-          profile_prefs->GetUserPrefValue(kDefaultModelKey)) {
-    if (base::EqualsCaseInsensitiveASCII(default_model_value->GetString(),
-                                         kDefaultModelBasicFrom)) {
-      profile_prefs->SetString(kDefaultModelKey, kDefaultModelBasicTo);
+  if (ai_chat::features::IsAIChatEnabled()) {
```

Comment:
we also need to add this condition inside RegisterProfilePrefsForMigration

Reply:
added